dc-cudami-editor/src/components/lists/PagedUserList.jsx

```diff
@@ -0,0 +1,173 @@
+import classNames from 'classnames'
+import React, {useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {FaHashtag, FaPowerOff, FaToggleOff, FaToggleOn} from 'react-icons/fa'
+import {Button, Card, CardBody, Col, Row, Table} from 'reactstrap'
+
+import {changeUserStatus, typeToEndpointMapping} from '../../api'
+import usePagination from '../../hooks/usePagination'
+import DeactiveUserDialog from '../dialogs/DeactiveUserDialog'
+import FeedbackMessage from '../FeedbackMessage'
+import Pagination from '../Pagination'
+import ActionButtons from './ActionButtons'
+
+const PagedUserList = ({apiContextPath = '/'}) => {
+  const type = 'user'
+  const {
+    content: users,
+    numberOfPages,
+    pageNumber,
+    pageSize,
+    totalElements,
+    setContent,
+    setPageNumber,
+  } = usePagination(apiContextPath, type)
+  const [dialogOpen, setDialogOpen] = useState(false)
+  const [feedbackMessage, setFeedbackMessage] = useState()
+  const [userIndex, setUserIndex] = useState()
+  const {t} = useTranslation()
+  const viewBaseUrl = `${apiContextPath}${typeToEndpointMapping[type]}`
+  const changeStatus = async (index = userIndex) => {
+    const user = users[index]
+    const successful = await changeUserStatus(
+      apiContextPath,
+      user.uuid,
+      !user.enabled
+    )
+    if (!successful) {
+      return setFeedbackMessage({
+        color: 'danger',
+        key: user.enabled
+          ? 'userNotDeactivatedSuccessfully'
+          : 'userNotActivatedSuccessfully',
+        values: {email: user.email},
+      })
+    }
+    const mutated = [...users]
```

Comment:
Warum packst Du das Array aus und packst es wieder in ein Array?

Reply:
Ich möchte das Originalarray nicht mutieren :smiley:

Reply:
Hab es entfernt.